src/main/java/billgates/database/MySQLDatabaseGateway.java

```diff
@@ -14,6 +14,8 @@
 
 public class MySQLDatabaseGateway implements DatabaseGateway {
     private Connection connection = null;
+
+    private int userId;
```

Comment:
In this case, the `userId` should be defined in the interface.

We always use the database gateway interface instead of the concrete implementation in other layers. If `userId` is defined in the class, we would not have access to modify it from the interface.